### PR TITLE
handle unkwnown new fields in API response

### PR DESCRIPTION
**Pull request by brad**
@orcasgit/orcas-developers Please review

- Refactor for maximum code reuse
- Upgrade some django code to not use deprecated APIs
- Remove `unique_together` constraint of the `Sleep` model, instead relying on a unique `id`
- Only retrieve 3 months of historic data, overridable with the `MISFIT_HISTORIC_TIMEDELTA` setting
- Add new fields to `Profile` and `Device` to reflect new data coming from the API
- Drop support for Django 1.4 & 1.6, add support for Django 1.9
- Drop support for Python 2.6 & 3.2, add support for Python 3.5
- Test upgraded dependencies

### CODE REVIEW

File: misfitapp/models.py [hunk 1]
```diff
@@ -44,29 +96,41 @@ class Meta:
         unique_together = ('user', 'date')
 
     @classmethod
-    def create_from_misfit(cls, misfit, uid, start_date=datetime.date(2014,1,1), end_date=datetime.date.today()):
+    def import_from_misfit(cls, misfit, uid, update=False,
```
Comment (grokcode) [inline misfitapp/models.py]: @brad I like this renaming. :+1:

Comment (brad) [inline misfitapp/models.py]: @grokcode :smile: Good, me too. For a while I was calling it `create_or_update_from_misfit` :open_mouth:

File: misfitapp/models.py [hunk 2]
```diff
@@ -2,17 +2,72 @@
...
+    d = dict((getattr(obj, field), obj) for obj in l)
+    return list(d.values())
+
+
+class MisfitModel(models.Model):
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def process_message(cls, message, misfit, uid):
```
Comment (grokcode) [inline misfitapp/models.py]: @brad Nice refactor here. I like having the models inherit from a base class and all use this logic.

File: misfitapp/tests/responses/goal_51a4189acf12e53f81000001.json [hunk 3]
```diff
@@ -2,5 +2,6 @@
     "id":"51a4189acf12e53f81000001",
     "date":"2014-10-05",
     "points":500,
-    "targetPoints":1000
+    "targetPoints":1000,
+    "unknown_new_field": "fakestuff"
```
Comment (grokcode) [inline misfitapp/tests/responses/goal_51a4189acf12e53f81000001.json]: @brad Thanks for testing that.

File: misfitapp/models.py (hunk 1 above)
Comment (grokcode) [inline misfitapp/models.py]: @brad Haha well that sounds a lot like what Django would call it.

**Pull request merged**